datatypes/src/operations/image/to_png.rs

```diff
@@ -0,0 +1,177 @@
+use crate::operations::image::{Colorizer, RgbaTransmutable};
+use crate::raster::{GridPixelAccess, Raster, Raster2D};
+use image::{DynamicImage, ImageFormat, RgbaImage};
+
+pub trait ToPng {
+    /// Outputs png bytes of an image of size width x height
+    fn to_png(&self, width: u32, height: u32, colorizer: &Colorizer) -> Vec<u8>;
+}
+
+impl<T> ToPng for Raster2D<T>
+where
+    T: Into<f64> + Copy + RgbaTransmutable,
+{
+    fn to_png(&self, width: u32, height: u32, colorizer: &Colorizer) -> Vec<u8> {
+        // TODO: use PNG color palette once it is available
+
+        let scale_x = (self.dimension()[0] as f64) / f64::from(width);
+        let scale_y = (self.dimension()[1] as f64) / f64::from(height);
+
+        let color_mapper = colorizer.create_color_mapper();
+
+        let image_buffer: RgbaImage = RgbaImage::from_fn(width, height, |x, y| {
+            // TODO: move these functions to base raster (?)
+            let cell_x = (((f64::from(x) + 0.5) * scale_x) - 0.5).round() as u32;
+            let cell_y = (((f64::from(y) + 0.5) * scale_y) - 0.5).round() as u32;
+
+            if let Ok(pixel_value) =
+                self.pixel_value_at_grid_index(&(cell_x as usize, cell_y as usize))
+            {
+                color_mapper.call(pixel_value)
+            } else {
+                colorizer.no_data_color()
+            }
+            .into()
+        });
+
+        let mut buffer = Vec::new();
+
+        let _ = DynamicImage::ImageRgba8(image_buffer).write_to(&mut buffer, ImageFormat::Png);
```

Comment:
what is the return value and why is it ok to ignore it?

Reply:
You are right, it is not.

4dfc595